### ISSUE TITLE
Discussion: a way to maintain API reference guides

### ISSUE THREAD

**Issue by jonas** [lightbend/paradox#26]
Using "API reference guides" for lack of a better word describing documentation which is tightly coupled to the implementation but higher level than Scaladoc and Javadoc and more like UNIX man pages containing both API and example code snippets etc.

A good example is Akka HTTPs directive documentation and the related ticket akka/akka-http#25:

 - http://doc.akka.io/docs/akka/2.4/scala/http/routing-dsl/directives/alphabetically.html

Scaladoc and Javadoc doesn't allow to easily organize and create such references. Furthermore, Scaladoc (up until 2.12) is not very search engine friendly, which is crucial for this type of documentation.

The question is if this is out of scope for Paradox itself since it would require some tight integration with scaladoc and/or javadoc generators. However this is also the case if cross references to source code is a goal as suggested in #25.

**Comment by eed3si9n** [comment]
> The question is if this is out of scope for Paradox itself since it would require some tight integration with scaladoc and/or javadoc generators.

If it helps Akka HTTP docs, it's in scope.

**Comment by ktoso** [comment]
Yes, some support for these things would help us a lot. We have around 300 directives which all have docs pages, and they're very similar to their javadoc + include an example.

We wanted to automate it in some way: 

 - https://github.com/akka/akka-http/issues/11
 - https://github.com/akka/akka-http/issues/10

because maintaining them is really hard.

**Comment by ktoso** [comment]
Basically what it is is "short description + long description + examples"
